pkg/apis/core/types.go

```diff
@@ -2543,12 +2543,6 @@ const (
 	PodReady PodConditionType = "Ready"
 	// PodInitialized means that all init containers in the pod have started successfully.
 	PodInitialized PodConditionType = "Initialized"
-	// PodReasonUnschedulable reason in PodScheduled PodCondition means that the scheduler
-	// can't schedule the pod right now, for example due to insufficient resources in the cluster.
-	PodReasonUnschedulable = "Unschedulable"
```

Comment:
Given that these are exported types in the core API, don't we need to mark them as deprecated first and then safely remove them?

Reply:
Generally we can suggest the users to use the literal from `k8s.io/api`, but not sure how to conventionally deprecate/remove this one (defined in k/k). Leaving the decision to sig-apimachinery.

Reply:
A think a more general question asking advice for sig-apimachinery is: inside and outside k/k, shall we restrict accessing const literals from k/api only? If so, we can remove duplicated const literals from k/k#pkg/apis/.

Reply:
weight in @liggitt for advices.

Reply:
all components other than the api-server should depend on the staging repo only (i.e., the external type), I think those can be removed if not used by the api-server.

Reply:
this variable is not used within k/k.

Reply:
cc @liggitt any concern here?

Reply:
the constants in the non-staging packages can be dropped if that makes sense, and we can switch things to reference the staging k8s.io/api exported constants

Reply:
Thanks, can you approve this? Needs api approval.